src/WebJobs.Script/Description/FunctionInvokerBase.cs

```diff
@@ -97,9 +99,10 @@ protected bool InitializeFileWatcherIfEnabled()
         {
             if (Host.ScriptConfig.FileWatchingEnabled)
             {
-                string functionScriptDirectory = Path.GetDirectoryName(Metadata.ScriptFile);
-                _fileWatcher = new AutoRecoveringFileSystemWatcher(functionScriptDirectory);
-                _fileWatcher.Changed += OnScriptFileChanged;
+                _fileChangeSubscription = Host.EventManager.OfType<FileEvent>()
+                    .Where(f => string.Equals(f.Source, "ScriptFiles", StringComparison.Ordinal) &&
+                    f.FileChangeArguments.Name.StartsWith(Metadata.Name, StringComparison.OrdinalIgnoreCase))
```

Comment:
Good catch. I'll make sure the function name is taken from the path (the actual directory name)